### PR TITLE
chore: added partition to common

### PR DESCRIPTION
**Pull request by daomry**
added partition() to common:
Partition unordered data into buckets of similar total weight

### CODE REVIEW

File: packages/patterns/src/index.ts [hunk 1]
```diff
@@ -10,3 +10,4 @@ export * from './lru-cache';
 export * from './set-multi-map';
 export * from './debouncer';
 export * from './disposables';
+export * from './partition';
```
Comment (Ari-Kishon) [inline packages/patterns/src/index.ts:13]: fix no eof

File: packages/patterns/src/partition.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/**
+ * Partition unordered data into buckets of similar total weight
+ *
+ * @example
+ * ```typescript
+ *  partition([1, 2, 3], 2, (i) => i)) // [[3], [2, 1]];
```
Comment (Ari-Kishon) [inline packages/patterns/src/partition.ts:6]: Could ```(i) => i``` be the default?

**Review by Ari-Kishon — APPROVED** [review]

File: packages/patterns/src/partition.ts (hunk 2 above)
Comment (daomry) [reply packages/patterns/src/partition.ts:6]: I think that's not interesting for numbers (other than examples and tests) usually you'd want to use the weight (as in test run time) to partition something else (test file name)

**Comment by daomry** [comment]
@Ari-Kishon this belongs in patterns package or common?

**Pull request merged**